[math] MovingAverage with long history

I'm trying to have a moving average with a long history:

<details>

<summary>

Details (click to see complete `main.cpp`)

```cpp
modm::filter::MovingAverage<int32_t, 100> average;
```

</summary>

```cpp
#include <modm/board.hpp>
#include <modm/math/filter.hpp>


using namespace Board;

int
main()
{
	Board::initialize();
	LedD13::setOutput();

	// Use the logging streams to print some messages.
	// Change MODM_LOG_LEVEL above to enable or disable these messages
	MODM_LOG_DEBUG   << "debug"   << modm::endl;
	MODM_LOG_INFO    << "info"    << modm::endl;
	MODM_LOG_WARNING << "warning" << modm::endl;
	MODM_LOG_ERROR   << "error"   << modm::endl;

	uint32_t counter(0);

	modm::filter::MovingAverage<int32_t, 100> average;

	while (true)
	{
		LedD13::toggle();
		modm::delay_ms(10);
		counter = counter <= 200 ? counter + 1 : (-counter);
		average.update(counter);
		MODM_LOG_INFO << "loop: " << average.getValue() << modm::endl;
	}

	return 0;
}
```

</details>

However, this doesn't compile and I get the following error from [`here`](https://github.com/modm-io/modm/blob/develop/src/modm/math/filter/moving_average.hpp#L104) which is caused by the [`least_uint` here](https://github.com/modm-io/modm/blob/develop/src/modm/math/utils/integer_traits.hpp#L28):
```
modm/src/modm/math/utils/integer_traits.hpp: In instantiation of 'struct modm::uint_t<100>':
modm/src/modm/math/utils/integer_traits.hpp:36:7:   required by substitution of 'template<int Bits> using least_uint = typename modm::uint_t::least [with int Bits = 100]'
modm/src/modm/math/filter/moving_average.hpp:104:16:   required from 'class modm::filter::MovingAverage<short int, 100>'
main.cpp:32:44:   required from here
modm/src/modm/math/utils/integer_traits.hpp:28:15: error: no type named 'type' in 'struct std::enable_if<false, long long unsigned int>'
   28 |         using least = std::conditional_t<
      |               ^~~~~
```

I would expect the `MovingAverage`'s class member `index` have an unsigned integer type, such that the maximum representable value by that type is greater than `N`. 

For simplicity, this could be `std::size_t` to cover all cases or `uint16_t` for most practical cases.

Or am I getting something wrong why `least_uint<N>` aka "Trait the smallest unsigneed type that fits n Bits" makes sense here?

--- a/src/modm/math/filter/moving_average.hpp
+++ b/src/modm/math/filter/moving_average.hpp
@@ -25,6 +25,7 @@
 #include <span>
 #include <algorithm>
 #include <numeric>
+#include <bit>
 
 #include <modm/math/utils/integer_traits.hpp>
 
@@ -101,7 +102,7 @@ public:
 	}
 
 private:
-	least_uint<N> index{0};
+	least_uint<std::bit_width(N)> index{0};
 	T buffer[N];
 	T sum;
 };

